presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -798,14 +802,17 @@ public static String annotateColumnComment(Optional<String> comment, boolean par
         return resultBuilder.build();
     }
 
-    public static String getPrefilledColumnValue(HiveColumnHandle columnHandle, HivePartitionKey partitionKey, Path path)
+    public static String getPrefilledColumnValue(HiveColumnHandle columnHandle, HivePartitionKey partitionKey, Path path, OptionalInt bucketNumber)
     {
         if (partitionKey != null) {
             return partitionKey.getValue();
         }
         if (isPathColumnHandle(columnHandle)) {
             return path.toString();
         }
+        if (isBucketNumberColumnHandle(columnHandle)) {
+            return bucketNumber.isPresent() ? Integer.toString(bucketNumber.getAsInt()) : null;
```

Comment:
Done. This is more safe on type: if type of argument is changed, another overload will be chosen, right?